### PR TITLE
fix: panic app when cannot send prove

### PR DESCRIPTION
**Pull request by toanbk**
- [x] Remove unused import
- [x] Fail gracefully during a temporary network disconnection.

### CODE REVIEW

File: clients/cli/src/prover.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ use nexus_core::{
         init_circuit_trace, key::CanonicalSerialize, pp::gen_vm_pp, prove_seq_step, types::*,
     },
 };
-use std::env;
+// use std::env;
```
Comment (collinjackson) [inline clients/cli/src/prover.rs:35]:
```suggestion
```

File: clients/cli/src/prover.rs [hunk 2]
```diff
@@ -125,10 +125,25 @@ async fn main() {
...
+            eprintln!("Failed to send message: {:?}, attempt {}/{}", e, retries + 1, max_retries);
+
+            retries += 1;
+            if retries >= max_retries {
+                eprintln!("Max retries reached, exiting...");
+                break;
+            }
+
+            // Optionally, add a delay before retrying
+            tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
```
Comment (collinjackson) [inline clients/cli/src/prover.rs:142]: Can we do exponential backoff here? (Multiply sleep delay by 2 each time.)

File: clients/cli/src/prover.rs [hunk 3]
```diff
@@ -261,10 +276,26 @@ async fn main() {
...
+                    eprintln!("Failed to send message: {:?}, attempt {}/{}", e, retries + 1, max_retries);
+        
+                    retries += 1;
+                    if retries >= max_retries {
+                        eprintln!("Max retries reached, exiting...");
+                        break;
+                    }
+        
+                    // Optionally, add a delay before retrying
+                    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
```
Comment (collinjackson) [inline clients/cli/src/prover.rs:293]: Can we do exponential backoff here? (Multiply sleep delay by 2 each time.)

**Review by collinjackson — COMMENTED** [review]
Looks great! Just a few minor comments, let me know if you agree and then I'll merge it

File: clients/cli/src/prover.rs (hunk 3 above)
Comment (toanbk) [reply clients/cli/src/prover.rs:293]: okay, let me update

Comment (collinjackson) [reply clients/cli/src/prover.rs:293]: -                    // Optionally, add a delay before retrying
-                    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+                    tokio::time::sleep(tokio::time::Duration::from_secs(u64::pow(2, retries))).await;

Comment (toanbk) [reply clients/cli/src/prover.rs]: I updated, please help me check and merge.

Thank you

**Pull request merged**